srml/contract/src/lib.rs

```diff
@@ -14,41 +14,64 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate. If not, see <http://www.gnu.org/licenses/>.
 
-//! Smart-contract module for runtime; Allows deployment and execution of smart-contracts
-//! expressed in WebAssembly.
+//! # Contract Module
+//! 
+//! The contract module provides functionality to deploy and execute smart-contracts expressed in WebAssembly.
+//! To use it in your module, you need to implement the contract [`Trait`].
+//! The supported dispatchable functions are documented as part of the [`Call`] enum.
+//! 
+//! ## Overview
+//! 
+//! This module provides the ability to create smart-contract accounts and send them messages.
```

Comment:
I am not saying this is not correct but just a bit different from my mental model (and the following sentence actually touches this a bit). I'd use this:

```suggestion
//! This module provides the ability to create smart-contracts and send them messages.
```

The substrate contracts module have one important difference from, say, the ethereum contract model: the contracts are not the first class citizen here and there is a whole SRML ecosystem which we need to integrate to.

One implication of this is the notion of an account is defined outside of the contract module and the contract module just adds some additional logic/data on top of it. I feel this is separation is stronger than, say, in ethereum model, where an account is something is defined on the same level. So because of that I've been using this terminology so far:

1. an account is an entity that have balance and identity. It is separate entity and a chain can have no contracts at all.
2. a contract is an extension of an account which exists only within the contract module. If there is no contract module then there is no contracts in a chain. An account can exist without a contract, a contract can't exist without an associated account though.

Reply:
cool - thanks for the clear explanation of the nuance here. will rephrase

Reply:
This module extends accounts with the ability to create smart-contracts and make calls to the contracts. Each smart-contract is associated with an account, which handles the code and storage. When a smart-contract is called, its associated code gets executed.
This code can alter the storage entries of the associated account, create new smart-contracts, or call other smart-contracts.